From 96907a392bcbad258541430f09e3bb568e58dc70 Mon Sep 17 00:00:00 2001
From: Arun Ravindran <arun.ravindran@intel.com>
Date: Wed, 28 Nov 2012 14:27:06 +0200
Subject: [PORT FROM MAIN] Telephony: Initialize Menu.defaultItem to -1

BZ: 63971

Valid value for defaultItem is from 0...n-1.
Initializing it to 0 will result in 1st item
in the menu selected even when the CAT hasn't
supplied any default item.

Initialize the defaultItem to -1 to avoid
the 1st item getting selected.

Change-Id: I26b19f0a0049fbcef40a638e1a6593d1d00681e2
Orig-Change-Id: I58b4e802aec7e8e7e0cfcb1f7a2941ec86ffa7dc
Category: aosp improvement
Domain: Telephony-Telephony-FWK
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../com/android/internal/telephony/cat/Menu.java   |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/java/com/android/internal/telephony/cat/Menu.java b/src/java/com/android/internal/telephony/cat/Menu.java
index a3cbbdf..2673cc3 100644
--- a/src/java/com/android/internal/telephony/cat/Menu.java
+++ b/src/java/com/android/internal/telephony/cat/Menu.java
@@ -44,7 +44,7 @@ public class Menu implements Parcelable {
         items = new ArrayList<Item>();
         title = null;
         titleAttrs = null;
-        defaultItem = 0;
+        defaultItem = -1;
         softKeyPreferred = false;
         helpAvailable = false;
         titleIconSelfExplanatory = false;
-- 
1.7.4.1

